File: Java/BinarySearchTree.java
public class BinarySearchTree {
    public class Node {
        int data;
        Node left;
        Node right;

        public Node(data, leftNode, rightNode){
            data = data;
            left = leftNode;
            right = rightNode;
        }
    }

    Node root;

    public BinarySearchTree(){
        root = null;
    }

    public void insert(int data){
        if (node != null){
            root = new Node(data, null, null);

        } else {
            insert(data, root);
        }
    }

    public Node insert(int data, Node node){
        if (node == null){
            node = Node(data, null, null);
        else if (data < node.data){
            return insert(data, node.left);
        } else {
            return insert(data, node.right);
        }
    }

    public bool search(int data){
        if (root == null) {
            System.out.println("Stack Empty");
            return false;
        } else {
            return search(data, root);
        }
    }

    public bool search(int data, Node node){
        if (node.data == data){
            return true;
        } else if (data < node.data){
            return search(data, node.left);
        } else {
            return search(data, node.right);
        }
    }
    
    // public void delete(){}
    // public void deleteNode(){}

    public int size()
    {
        if (root == null){
            return 0;
        } else {
            return size(root.left) + size(root.right) + 1;
        }
    }

    public int size(Node Node){
        if (node == null){
            return 0;
        } else {
            return size(root.left) + size(root.right) + 1;
        }
    }


    public void preOrder(Node node){
        if (node != null){
            Console.WriteLine(node.data);
            preOrder(node.left);
            preOrder(node.right);
        }
    }

    public void postOrder(Node Node){
        if (node != null){
            postOrder(node.left);
            postOrder(node.right);
            Console.WriteLine(node.data);
        }
    }

    public void inOrder(Node node){
        if (node != null){
            inOrder(node.left);
            System.out.println(node.data);
            inOrder(node.right);
        }
    }

    public static void main(string [] args) {
        BinarySearchTree binarysearchtree = new BinarySearchTree();
    }
}

File: PHP/BinarySearchTree.php
<?php
class Node {
    public $data;
    public $left;
    public $right;
    
    function __contruct($data, $leftNode, $rightNode){
        $this->data = $data;
        $this->left = $leftNode;
        $this->right = $rightNode;
    }

    function setLink($node){
        $this->link = $node;
    }

    function getLink(){
        return $this->link;
    }
}

class BinarySearchTree {
    public $root;

    function __contruct(){
        $this->root = null;
    }

    function insert($data){
        if ($data == null){
            $this->root = new Node($data, null, null);
        } else {
            insertNode($data, $this->root);
        }
    }

    function insertNode($data, $node){
        if ($node == null){
            $node = new Node($data, null, null);
        } else if ($data < $node->data){
            insertNode($data, $node->left);
        } else {
            insertNode($data, $node->right);
        }
    }
    
    // function delete(){}
    // function deleteNode($data){}

    function find($data){
        if ($this->root == null){
            echo "Binary Search Tree Empty !"
            return false;
        } else {
            return findNode($data, $this->root);
        }
    }

    function findNode($data, $node){
        if ($data == $node->data){
            return true;
        } else if ($data < $node->data){
            return findNode($data, $node->left);
        } else {
            return findNode($data, $node->right);
        }
    }

    function size(){
        if ($this->root == null){
            return 0;
        } else {
            return sizeNode($this->root) + sizeNode($this->root) + 1;
        }
    }

    function sizeNode($node){
        if ($node == null){
            return 0;
        } else {
            return sizeNode($this->root) + sizeNode($this->root) + 1;
        }
    }

    function preOrder($node){
        if ($node != null){
            echo $node->data;
            preOrder($node->left);
            preOrder($node->right);
        }
    }

    function postorder(){
        if ($node != null){
            postOrder($node->left);
            postOrder($node->right);
            echo $node->data;
        }
    }
    
    function inOrder(){
        if ($node != null){
            preOrder($node->left);
            echo $node->data;
            preOrder($node->right);
        }
    }
}

$binarysearchtree = new BinarySearchTree();


?>
